### PR TITLE
Replace re-frame with signum

### PR DESCRIPTION
**Pull request by sandhu**

### CODE REVIEW

File: example/src/via/example/core.cljs [hunk 1]
```diff
@@ -1,16 +1,11 @@
...
-  (:require [via.example.app :refer [app]]
-            [via.example.events]
-            [via.example.subs]
+  (:require [via.example.state]
+            [via.example.app :refer [app]]
             [via.example.views :as views]
-            [reagent.dom :as reagent]
-            [re-frame.core :as re-frame]))
+            [helix.core :as hx :refer [$]]
+            ["react-dom" :as react-dom]))
```
Comment (tom7t) [inline example/src/via/example/core.cljs:6]: This can just be a symbol instead of a string.

File: example/src/via/example/state.cljs [hunk 2]
```diff
@@ -0,0 +1,62 @@
+;;   Copyright (c) 7theta. All rights reserved.
+;;   The use and distribution terms for this software are covered by the
+;;   MIT License (https://opensource.org/licenses/MIT) which can also be
+;;   found in the LICENSE file at the root of this distribution.
+;;
+;;   By using this software in any fashion, you are agreeing to be bound by
+;;   the terms of this license.
+;;   You must not remove this notice, or any others, from this software.
+
+(ns via.example.state
```
Comment (tom7t) [inline example/src/via/example/state.cljs:10]: Nice, liking this approach.

File: example/src/via/example/macros.cljc [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+#?(:clj
+   (defmacro defnc [type params & body]
+     (let [opts? (map? (first body))
+           opts (if opts? (first body) {})
+           body (if opts? (rest body) body)
+           default-opts {:helix/features {:fast-refresh true}}]
+       `(helix.core/defnc ~type ~params
+          ~(merge default-opts opts)
+          ~@body))))
```
[the remark this answers is not in the document]
Comment (tom7t) [reply example/src/via/example/macros.cljc:13]: I guess it's not too bad if the re-frame-template (name change needed?) project handles this.

**Review by tom7t — APPROVED** [review]

File: example/src/via/example/macros.cljc (hunk 3 above)
[the remark this answers is not in the document]
Comment (sandhu) [reply example/src/via/example/macros.cljc:13]: Yeah. Been debating the same thing. I really belongs in a "UX Library" of some kind, but for now, I'll create a new template and put it there.

File: example/src/via/example/core.cljs [hunk 4]
```diff
@@ -1,16 +1,11 @@
...
-  (re-frame/clear-subscription-cache!)
-  (reagent/render [views/main-panel] (.getElementById js/document "app")))
-
-(defn ^:export init []
-  (re-frame/dispatch-sync [:initialize-db])
+(defn ^:export init
+  []
   (enable-console-print!)
-  (mount-root))
+  (react-dom/render ($ views/main-panel) (js/document.getElementById "app")))
```
Comment (tom7t) [inline example/src/via/example/core.cljs:11]: Should there be an `:as` alias in the require? If not, I didn't realize that would work. That's cool.

**Review by tom7t — APPROVED** [review]

File: example/src/via/example/core.cljs (hunk 4 above)
Comment (sandhu) [reply example/src/via/example/core.cljs:11]: Since we're not "aliasing" `react-dom`, the `:as` is not required

**Pull request merged**